core/src/main/java/org/apache/cxf/io/DelayedCachedOutputStreamCleaner.java

```diff
@@ -74,25 +78,146 @@ default void forceClean() {
+                    timerTask = null;
+                }
+            }
+            if (oldTimerTask != null) {
+                oldTimerTask.cancel();
+            }
+        }
+
+        @Override
+        public void clean() {
+            final Collection<DelayedCloseable> closeables = new ArrayList<>();
+            TimerTask oldTimerTask = null;
+            synchronized (lock) {
+                queue.drainTo(closeables);
+                if (queue.isEmpty() && timerTask != null) {
+                    oldTimerTask = timerTask;
+                    timerTask = null;
+                }
+            }
+            if (oldTimerTask != null) {
+                oldTimerTask.cancel();
+            }
+            clean(closeables);
+        }
+
+        @Override
+        public void forceClean() {
+            TimerTask oldTimerTask = null;
+            synchronized (lock) {
+                clean(queue);
+                if (timerTask != null) {
+                    oldTimerTask = timerTask;
+                    timerTask = null;
+                }
+            }
+            if (oldTimerTask != null) {
+                oldTimerTask.cancel();
+            }
+        }
+
+        @Override
+        public void close()  {
+            TimerTask oldTimerTask = null;
+            synchronized (lock) {
+                queue.clear();
+                if (timerTask != null) {
+                    oldTimerTask = timerTask;
+                    timerTask = null;
+                }
+            }
+            if (oldTimerTask != null) {
+                oldTimerTask.cancel();
+            }
+        }
+
+        @Override
+        public int size() {
+            return queue.size();
+        }
+
+        private void clean(Collection<DelayedCloseable> closeables) {
+            final Iterator<DelayedCloseable> iterator = closeables.iterator();
+            while (iterator.hasNext()) {
+                final DelayedCloseable next = iterator.next();
+                try {
+                    iterator.remove();
+                    LOG.warning("Unclosed (leaked?) stream detected: " + next.closeable.hashCode());
+                    next.closeable.close();
+                } catch (final IOException | RuntimeException ex) {
+                    LOG.warning("Unable to close (leaked?) stream: " + ex.getMessage());
+                }
+            }
+        }
+    }
+
+    private static final class DefaultDelayedCleaner implements DelayedCleaner {
         private final long delay; /* default is 30 minutes, in milliseconds */
         private final DelayQueue<DelayedCloseable> queue = new DelayQueue<>();
         private final Timer timer;
-        
-        DelayedCleanerImpl(final long delay) {
+        private final AtomicBoolean scheduled = new AtomicBoolean(false);
+
+        DefaultDelayedCleaner(final long delay) {
             this.delay = delay;
             this.timer = new Timer("DelayedCachedOutputStreamCleaner", true);
```

Comment:
Constructing the timer already starts the Timer thread to sit in a wait state for the first task to be scheduled so this breaks the lazy thread creation.